### DISCUSSION TITLE
VS 2022 Preview IntelliSense Coming to VS Code?

### DISCUSSION THREAD

**Discussion by mikeseese** [microsoft/vscode-cpptools#8365] [General]
Hello,

There was an [article](https://devblogs.microsoft.com/cppblog/18x-faster-intellisense-for-unreal-engine-projects-in-visual-studio-2022/) posted roughly a week ago boasting an 18x performance boost for IntelliSense in Unreal Engine projects. This is huge, but I prefer using VS Code over VS whenever I possibly can. I was wondering if/when the IntelliSense improvements would make it to this extension, even as part of a beta or insider build?

Thanks!

**Comment by sean-mcmanus** [comment]
I believe the performance improvements would require us to add support for https://github.com/microsoft/vscode-cpptools/issues/3628 and https://github.com/microsoft/vscode-cpptools/issues/8355 . It's on our TODO list and we're actively working on unifying the VS and VS Code code bases to allow for easier cross platform/IDE feature sharing.

**Reply by mikeseese** [reply]
Are these tasks that outside contributors can work on (i.e. are part of this open codebase) or are these features that need to be implemented in the closed source libraries that this extension uses?

**Reply by sean-mcmanus** [reply]
Closed source part, shared with VS.

**Comment by mikeseese** [comment]
Hey @sean-mcmanus, I just wanted to touch base and mention that this is still a highly desired feature that affects me daily with developing an Unreal Engine project within VSCode. I believe it's a blocker for anyone else from even considering to use VSCode for C++ Unreal development, as I need to use clang to get reasonable performance (and that still sucks). Are there any plans to get #3628 promoted off the backlog? It's a bummer that it's part of the closed source part as I'd love to help if possible.

**Reply by sean-mcmanus** [reply]
I think it'll get covered as part of our long-running unification efforts, but this particular issue isn't on deck yet.

@bobbrow Did you have anything to add?
